okio-kotlin/src/main/java/okio/ByteString.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("NOTHING_TO_INLINE") // Aliases to public API.
+
+package okio
+
+import java.io.InputStream
+import java.nio.ByteBuffer
+import java.nio.charset.Charset
+import kotlin.text.Charsets.UTF_8
+
+/**
+ * Returns a [ByteString] containing a copy of this [ByteArray].
+ *
+ * @see ByteString.of
+ */
+inline fun ByteArray.toByteString(offset: Int = 0, byteCount: Int = size): ByteString =
+    ByteString.of(this, offset, byteCount)
+
+/**
+ * Returns a [ByteString] containing a copy of this [ByteBuffer].
+ *
+ * @see ByteString.of
+ */
+inline fun ByteBuffer.toByteString(): ByteString = ByteString.of(this)
+
+/**
+ * Returns a [ByteString] containing the [charset]-encoded bytes of this [String].
+ *
+ * @see ByteString.encodeString
+ */
+inline fun String.encodeByteString(charset: Charset = UTF_8): ByteString =
+    ByteString.encodeString(this, charset)
```

Comment:
If the charset is UTF_8, is it worth doing ByteString.encodeUtf8? (or have a separate function for utf8?)

Reply:
Both have the same implementation, calling `string.getBytes(charset)`.

Reply:
That optimization should live inside of encodeString if we want to make it.

Reply:
Yeah, I guess I meant to ask if it's worth having a String.encodeUtf8ByteString() extension here.

Reply:
I don't think so. That actively harms the Kotlin API. I would rather see the optimization encapsulated behind the Java API.